File: 2.Recursion/j.power-with-recursion/power-with-recursion.cpp
#include <iostream>

int power(int m,int n)
{
    if(n == 0)
        return 1;
    else
        return power(m, n-1) * m;
}

int main()
{
    int n = 2;
    int m = 25;

    std::cout << "Power Recursion method -> power("<<m <<","<<n<<") : "<<power(m,n)<<std::endl;

    return EXIT_SUCCESS;
}

File: 2.Recursion/j.power-with-recursion/go/power-with-recursion.go
package main

import "fmt"

func power(m int, n int) int {
	if n == 0 {
		return 1;
	}else {
		return  power(m,n-1) * m;
	}
}

func main(){
	m,n := 9,3;

	fmt.Printf("Recurion value for power(%d,%d) : %d",m,n,power(m,n)); 
}